### ISSUE TITLE
CRITIAL : Error · FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: Trying to deserialize version that isn't set

### ISSUE THREAD

**Issue by sambegin** [firebase/firebase-js-sdk#1339]
### [REQUIRED] Describe your environment

  * Operating System version:  iOS 11 and iOS 12 
  * Firebase SDK version: latest (5.5.6)
  * Firebase Product: Firestore

### Describe the problem
We have hundreds of users that cannot even use the app because we have this error. 

#### Steps to reproduce:
Only open a couple of snapshot... Then this error is popping in console and no get(), set() or update() is working after that.   

#### Relevant Code:
There is no code I can show you to reproduce that, it seems to be an internal library error and we just need to know why this is happening. Just knowing the reason this error is raised could help us find the problem. Thanks

**Comment by sambegin** [comment]
We have persistence enabled. 
We have multi-tab enabled. ({ experimentalTabSynchronization: true }). This error occurred with version 5.5.0 and 5.5.6 now. 

Looking at our logs right now it seems to only happen on iOS devices (we have a cordova app so same code is used between android and iOS users, but 95 % of our users are on iOS) 

We did not see this on browsers because most of our users is on mobile devices so I don't know if multitab flag is in play in this case. If you need anything else let me know

**Comment by wilhuff** [comment]
If you've observed this yourself a run with logging enabled would be helpful: https://firebase.google.com/docs/reference/js/firebase.firestore.Firestore#setLogLevel

Was this working for you prior to 5.5.0? If you downgrade does the problem go away?

Basically anything you can do to help narrow this down would be helpful.

**Comment by sengoontoh** [comment]
I'm facing the same issue as well. Unfortunately unable to reproduce it but I'm seeing it in the field.

**Comment by wilhuff** [comment]
When you say you’re seeing this as well, is it the identical circumstances, or are you seeing this in non-Cordova settings?

**Comment by jggc** [comment]
I can't speak for sengoontoh but we are experiencing this issue on Cordova iOS and Safari iOS at the moment.

Here is a StackTrace from Mobile Safari 11.4.1

![screenshot_2018-10-29 medyx timeline 1](https://user-images.githubusercontent.com/6208924/47727394-b9e3f500-dc32-11e8-8a26-afe094e94c05.png)

**Comment by sengoontoh** [comment]
> I'm facing the same issue as well. Unfortunately unable to reproduce it but I'm seeing it in the field.

I'm seeing it on Cordova ios 11.4.1, 12.0, and 12.0.1 as well as Cordova android 8.0.

It seems to happen when the app is first launched.

**Comment by sambegin** [comment]
We are effectively receiving this error 
` FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: Got result for empty write pipeline` 
along with with the error in this issue

**Comment by swftvsn** [comment]
We're also seeing this on production now, it affects also desktop browsers, atleast chrome and safari. Latest error was sent from (according to sentry.io)

Chrome, Version: 69.0.3497
Mac OS X, Version: 10.13.6

**Comment by swftvsn** [comment]
And it's occurring only for handful of users, quite low percentage for us, so I've not been able to repro yet.

**Comment by sengoontoh** [comment]
Are yall seeing these errors as well?

FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: Got result for empty write pipeline

**Comment by sambegin** [comment]
@sengoontoh  Yes for us. We have about 35 % "pipeline" errors and 65% "deserialize" errors

**Comment by wilhuff** [comment]
Just to follow up for those that aren't working with our support org: this is a Firestore-wide incident: https://status.firebase.google.com/incident/Cloud%20Firestore/18008.

All versions of the JS SDK are affected. We're in the process of rolling back the backend release that that introduced the issue.

**Comment by wilhuff** [comment]
Rollback is complete.

@sambegin, @sengoontoh, @jggc, @swftvsn could you confirm that you're no longer seeing issues?

**Comment by sambegin** [comment]
@wilhuff  

Our users doesn't use the app very much at this time of the day so I'll be able to confirm it later or tomorrow. I've not seen the error since about more than an hour ago in our logs.

**Comment by sengoontoh** [comment]
@wilhuff 

I just saw an error at 3:39:43.088 PM GMT

FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: Got result for empty write pipeline

**Comment by sengoontoh** [comment]
@wilhuff 

Got a FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: Trying to deserialize version that isn't set

at 3:52 pm PST

**Comment by sengoontoh** [comment]
@wilhuff 

Another deserialize INTERNAL ASSERTION FAILED at 4:12 pm PST.

**Comment by wsh** [comment]
Hi all--mea culpa. We're still doing the rollback, I ran a query incorrectly and jumped the gun. We expect this to complete within the hour, and I'll let you know when it has.

**Comment by wilhuff** [comment]
Sigh. It turns out the rollback was only mostly complete. I’ll keep you posted.

**Comment by wsh** [comment]
The rollback is now complete, folks. We do expect this to have fixed the issue, so please let us know if it reoccurs!

**Comment by jggc** [comment]
Last occurence of this error at 4:17pm EST yesterday, almot 14 hours ago. But usage of our app is lower outside business hours so i'd wait 24h more before calling it over.

**Comment by sengoontoh** [comment]
I'm still seeing these errors as of 3 AM PST today.

**Comment by jggc** [comment]
@sengoontoh did you see any decrease in occurence since the rollback completed?

**Comment by sengoontoh** [comment]
Doesn't feel like it. It's also really early right now though so most of my users are not on yet.

**Comment by sengoontoh** [comment]
I'm still seeing these errors trickle in. One in the past hour. @sambegin, @jggc, @swftvsn  are you still seeing it?

**Comment by sambegin** [comment]
@sengoontoh  Not on our side, the last one we've seen was 21h ago.

**Comment by swftvsn** [comment]
No errors here since the rollback.

**Comment by yankedev** [comment]
My application has persistence enabled, but only on single tab. 
The error continued to appear today using a browser with data stored locally.
With Incognito window everything worked fine.

To fix the problem everywhere I have done:
In Chrome > Developer Tools > Application > delete the IndexedDB

**Comment by mikelehen** [comment]
@yankedev Can you confirm exactly what error you were seeing?  Thanks.

**Comment by youngnacs** [comment]
I got the same error when I tried to sign in via google firebase, which is 5.5.6 but error message is 👍 Error: FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: AsyncQueue is already failed: Internal error opening backing store for indexedDB.open.
auth.service.ts:96
ERROR
core.js:1673
Error: Uncaught (in promise): Error: FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: AsyncQueue is already failed: Internal error opening backing store for indexedDB.open.
core.js:1673
[2019-02-09T17:09:42.258Z]  @firebase/firestore: Firestore (5.5.0): FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: AsyncQueue is already failed: Internal error opening backing store for indexedDB.open.
index.esm.js:77
[[StableObjectId]]:2
message:"FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: AsyncQueue is already failed: Internal error opening backing store for indexedDB.open."
stack:"Error: FIRESTORE (5.5.0) INTERNAL ASSERTION FAILED: AsyncQueue is already failed: Internal error opening backing store for indexedDB.open.
